server/etcdserver/server.go

```diff
@@ -818,7 +818,13 @@ func (s *EtcdServer) run() {
 		snapi:     sn.Metadata.Index,
 		appliedt:  sn.Metadata.Term,
 		appliedi:  sn.Metadata.Index,
-		compacti:  fi - 1,
+		// compacti is the index from the last time raftStorage.Compact was called
+		// without errors.
+		//
+		// After calling raftStorage.Compact(compacti) without errors, the dummy entry of
+		// raftStorage becomes {Index: compacti}, and raftStorage.FirstIndex() returns
+		// (compacti+1, nil). This is validated by TestMemoryStorageCompaction.
```

Comment:
```suggestion
		// Compaction is inclusive, meaning compact index should be lower by one than the first index after compaction. 
		// This is validated by TestMemoryStorageCompaction.
```
Don't describe the variables in comments, if you want to make `compacti` more readable, then make the variable names more descriptive, but let's not address it now.